pkg/webhook/server.go

```diff
@@ -109,6 +117,17 @@ func (s *Server) setDefaults() {
 	if len(s.KeyName) == 0 {
 		s.KeyName = "tls.key"
 	}
+
+	switch s.TLSMinVersion {
+	case "1.1":
+		s.tlsMinVersion = tls.VersionTLS11
+	case "1.2":
+		s.tlsMinVersion = tls.VersionTLS12
+	case "1.3":
+		s.tlsMinVersion = tls.VersionTLS13
+	default:
```

Comment:
IMHO we should error if `TLSMinVersion` is set but not in the list, silently picking an old version is not a good idea IMHO

Reply:
Do you mind suggesting a mechanism for this (setDefaults doesn't currently have any error handling so I'm not sure what pattern to follow)

Reply:
I would just move this into `Start`. The only thing that really qualifies as defaulting is to set it to `tls.VersionTLS10` if s.TLSMinVersion is an empty string

Reply:
thanks for that suggestion - hopefully this iteration does what you're after.